frontend/packages/knative-plugin/src/components/test-function/modal-components/RequestBody.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { FormikProps, FormikValues } from 'formik/dist/types';
+import { YAMLEditorField } from '@console/shared/src';
+import { InvokeFormat } from '../../../types';
+
+const RequestBody: React.FC<FormikProps<FormikValues>> = ({ values }) => {
+  const {
+    request: {
+      headers: {
+        [InvokeFormat.CloudEvent]: { ContentType: CloudEventContentType },
+        [InvokeFormat.HTTP]: { ContentType: HTTPContentType },
+      },
+      format: invokeFormat,
+    },
+  } = values;
+
+  const getcurrentLanguage = () => {
+    if (invokeFormat === InvokeFormat.HTTP && HTTPContentType === 'application/json') {
+      return 'json';
+    }
+    if (invokeFormat === InvokeFormat.CloudEvent && CloudEventContentType === 'application/json') {
+      return 'json';
+    }
+    return 'yaml';
+  };
+
+  return (
+    <>
+      <div className="kn-test-sf-modal__editor">
+        <YAMLEditorField
```

Comment:
Maybe its time to rename the YAMLEditorField to CodeEditorField or just EditorField?

Reply:
20 Files will be changed. Will do it in the next PR.